plugins/modules/asset.py

```diff
@@ -108,15 +125,15 @@ def main():
     client = arguments.get_sensu_client(module.params["auth"])
     path = "/assets/{0}".format(module.params["name"])
     payload = arguments.get_mutation_payload(
-        module.params, "download_url", "sha512", "filters", "headers",
+        module.params, "url", "sha512", "filters", "headers",
     )
-    payload["url"] = payload.pop("download_url")  # Remap download_url -> url
+    payload = get_api_payload(payload)
```

Comment:
It feels weird to have "get_mutation_payload" and "get_api_payload" separated. Couldn't we somehow unify this into a single function so that we get rid of "create payload then pop from it to build actual payload" because when you read the code it's like "okay so what was the initial 'payload' then, semi-payload" 😄 ?

Something like:

```python
def build_api_payload(params):
  payload = arguments.get_mutation_payload(params)  # NOTICE: no flat parameters here, so we won't need to pop them later
  payload["builds"] = [dict(
    url=params.get("url"),
    sha512=params.get("sha512"),
    ...
  )]
  return payload
```